ui-speedspacechart/src/components/helpers/drawElements/reticle.ts

```diff
@@ -0,0 +1,249 @@
+        (data) => data.start <= previousCurvePosition.position
+      );
+
+      modeText =
+        previousElectrification?.electrificationUsage.object_type === 'Electrified'
+          ? 'electric'
+          : '--';
+
+      if (previousElectrification?.electrificationUsage.object_type === 'Electrified') {
+        electricalProfileText = `${previousElectrification?.electrificationUsage.mode} --`;
+      } else {
+        electricalProfileText = `${previousElectrification?.electrificationUsage.object_type} --`;
+      }
+
+      // find out if the cursor isn't close to the previous stop or the next stop based on 20px
+      if (
+        curveX - xPositionReference(previousStop.position) > 20 &&
+        xPositionReference(nextStop.position) - curveX > 20
+      ) {
+        const drop = (y.b! - y.a!) / (x.b! - x.a!);
+        const deltaX = curveX - x.a;
+        const deltaY = drop * deltaX;
+        curveY = y.a! + deltaY;
+
+        marecoSpeedText = (
+          previousCurvePosition.speed +
+            ((nextCurvePosition.speed - previousCurvePosition.speed) * (curveX - x.a)) /
+              (x.b - x.a) || 0
+        ).toFixed(1);
+
+        if (previousCurvePosition.speed < nextCurvePosition.speed) {
+          effortText = 'accelerating';
+        }
+        if (previousCurvePosition.speed > nextCurvePosition.speed) {
+          effortText = 'decelerating';
+        }
+
+        previousGradientText = store.slopes.findLast(
+          (data: GradientPosition) => xPositionReference(data.position) <= curveX!
+        )!.gradient;
+      } else {
+        // find out from wich side the cursor is closer to the stop
+        if (
+          curveX - xPositionReference(previousStop.position) <=
+          xPositionReference(nextStop.position) - curveX
+        ) {
+          curveX = xPositionReference(previousStop.position);
+        } else {
+          curveX = xPositionReference(nextStop.position);
+        }
+
+        const nextSpeed = store.speed.findLast(
+          (speed: ConsolidatedPositionSpeedTime) => xPositionReference(speed.position) <= curveX
+        )!.speed;
+        const previousSpeed = store.speed.find(
+          (speed: ConsolidatedPositionSpeedTime) => xPositionReference(speed.position) >= curveX
+        )!.speed;
+
+        // find curveY based on the average speed between the previous and next speed values
+        curveY =
+          cursorBoxHeight -
+          (((previousSpeed + nextSpeed) / 2 - minSpeed) / speedRange) *
+            (cursorBoxHeight - CURVE_MARGIN_TOP);
+
+        marecoSpeedText = ((previousSpeed + nextSpeed) / 2).toFixed(1);
+
+        if (
+          store.speed.findLast(
+            (speed: ConsolidatedPositionSpeedTime) => xPositionReference(speed.position) <= curveX
+          )!.speed <
+          store.speed.find(
+            (speed: ConsolidatedPositionSpeedTime) => xPositionReference(speed.position) >= curveX
+          )!.speed
+        ) {
```

Comment:
```suggestion
        if ( nextSpeed < previousSpeed ) {
```
Why not reuse nextSpeed and previousSpeed here?

Reply:
obviously... done